internal/extsvc/bitbucketcloud/client.go

```diff
@@ -265,22 +265,22 @@ func (c *client) do(ctx context.Context, req *http.Request, result any) error {
 
 	bs, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return err
+		return err, code
 	}
 
-	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
+	if code < 200 || code >= 400 {
```

Comment:
Probably a cleanup opportunity to not use hardcoded numbers for the statuses in here, too:

```suggestion
	if code < http.StatusOK || code >= http.StatusBadRequest {
```